File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.js
const isPrime=(number)=>{
    if(number===1||number===0) return false
    for(let i=2;i<number;i++){
        if(number%i===0) return false
    }
    return true
}

const permutation=(arr,n)=>{
    if(n===1) return arr.map(v=>[v])
    
    const answer=[]
    
    arr.forEach((v,idx,arr)=>{
        const rest=[...arr.slice(0,idx),...arr.slice(idx+1)]
        const permutations=permutation(rest,n-1)
        const attach=permutations.map(permutation=>[...permutation,v])
        answer.push(...attach)
    })
    return answer
}

function solution(numbers) {
   const cnt=new Set()
    for(let i=1;i<=numbers.length;i++){
        const filteredNumber=permutation(numbers.split(''),i).map(v=>Number(v.join(''))).filter(v=>isPrime(v)).forEach(v=>cnt.add(v))
    }
    return cnt.size
}

File: 백준/Bronze/1978. 소수 찾기/소수 찾기.cc
#include <iostream>

using namespace std;

bool isPrime(int N)
{
    if(N==1) return false;
    for(int i=2;i<N;i++){
        if(N%i==0) return false;
    }
    return true;
}

int main()
{
    int N;
    cin >> N;

    int cnt=0;
    for(int i=0;i<N;i++){
        int n;
        cin >> n;
        if(isPrime(n)) cnt++;
    }
    cout << cnt;
}